src/EditorFeatures/CSharpTest2/Recommendations/GlobalKeywordRecommenderTests.cs

```diff
@@ -170,5 +170,145 @@ class C
 {
     delegate*$$");
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestInCompilationUnit()
+        {
+            await VerifyKeywordAsync(@"
+$$
+");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestAfterExtern()
+        {
+            await VerifyKeywordAsync(
+@"extern alias goo;
+$$");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestAfterPreviousUsing()
+        {
+            await VerifyKeywordAsync(
+@"using Goo;
+$$");
```

Comment:
this technically isn't legal right?  global usings need to come before normal usings?  Or do we allow them to come after?

Note: even if this is not legal, i'm fine with the keyword recommender offering this.

Reply:
> this technically isn't legal right?  global usings need to come before normal usings?  Or do we allow them to come after?

> Note: even if this is not legal, i'm fine with the keyword recommender offering this.

You are correct. It is not legal, but I think it is better to offer it and report an error, than to not offer it. 

---
In reply to: [599780626](https://github.com/dotnet/roslyn/pull/52059#discussion_r599780626) [](ancestors = 599780626)

Reply:
> but I think it is better to offer it and report an error, than to not offer it.

Agreed.  I just wanted to make sure my mental model was correct.  In the impl it would be good to have a comment stating that in the place where we make the decisino just to make the intent clear.  Up to you.

Reply:
> In the impl it would be good to have a comment stating that in the place where we make the decisino just to make the intent clear.  Up to you.

There isn't really a logict to specifically allow that. We simply do not make any effort to disallow. 

---
In reply to: [599801190](https://github.com/dotnet/roslyn/pull/52059#discussion_r599801190) [](ancestors = 599801190)